src/ADAL.NET/UserCredential.cs

```diff
@@ -59,5 +59,16 @@ internal char[] PasswordToCharArray()
 
             return (this.Password != null) ? this.Password.ToCharArray() : null;
         }
+
+        internal char[] EscapedPasswordToCharArray()
+        {
+            string password = this.Password;
+            password = password.Replace("&", "&amp;");
+            password = password.Replace("\"", "&quot;");
+            password = password.Replace("'", "&apos;");
+            password = password.Replace("<", "&lt;");
+            password = password.Replace(">", "&gt;");
+            return (password != null) ? password.ToCharArray() : null;
```

Comment:
Then isn't there any other method to do the right thing? Also, is the list of characters you replace above complete?